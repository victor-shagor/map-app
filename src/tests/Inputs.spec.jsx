import { fireEvent, render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

import Input from "components/Inputs";

describe("input Component", () => {
  it("render input component", () => {
    render(<Input />);
    expect(screen.getByRole("textbox")).toBeInTheDocument();
  });

  it("onchange event should be triggered", () => {
    const onChange = jest.fn();

    render(<Input onChange={onChange} />);
    const input = screen.getByRole("textbox");
    expect(onChange).toHaveBeenCalledTimes(0);
    fireEvent.change(input, { target: { value: 20 } });
    expect(onChange).toHaveBeenCalled();
  });

  it("renders correctly", () => {
    const onChange = jest.fn();
    const tree = renderer.create(<Input onChange={onChange} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
