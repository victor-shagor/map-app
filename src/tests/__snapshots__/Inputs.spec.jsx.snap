// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input Component renders correctly 1`] = `
<div
  className="input-div"
>
  <label />
  <input
    className="input"
    onChange={[MockFunction]}
    required={true}
    step="any"
  />
</div>
`;
